extensions/rules.go

```diff
@@ -21,8 +21,8 @@ type Rule struct {
 	ID string
 	// Description defines description of what the rule does.
 	Description string
-	// Parameter is optional information for tuning rule logic.
-	Parameter *Parameter
+	// Parameters are optional information for tuning rule logic.
+	Parameters *[]Parameter
```

Comment:
Since we don't have to be concerns with encoding and decoding for this struct, I don't think it needs to be a pointer to a slice. WDYT?